paperqa/types.py

```diff
@@ -686,6 +686,16 @@ def __add__(self, other: DocDetails | int) -> DocDetails:  # noqa: PLR0912
                 merged_data[field] = {**self.other, **other.other}
                 # handle the bibtex / sources as special fields
                 for field_to_combine in ("bibtex_source", "client_source"):
+                    # Ensure the fields are lists before combining
+                    if self.other.get(field_to_combine) and not isinstance(
+                        self.other[field_to_combine], list
+                    ):
+                        self.other[field_to_combine] = [self.other[field_to_combine]]
+                    if other.other.get(field_to_combine) and not isinstance(
```

Comment:
Why do we have the same `if` statement twice? Can you either remove the dupe, or if it matters, add a comment explaining why?